cmd/local-artifact-mirror/serve.go

```diff
@@ -60,6 +64,34 @@ var serveCommand = &cli.Command{
 	},
 }
 
+// startBinaryWatcher starts a loop that observes the binary until its modification
+// time changes. When the modification time changes a SIGTERM is send in the provided
+// channel.
+func startBinaryWatcher(stop chan os.Signal) error {
+	fpath := defaults.PathToEmbeddedClusterBinary("local-artifact-mirror")
+	stat, err := os.Stat(fpath)
+	if err != nil {
+		return fmt.Errorf("unable to stat %s: %s", fpath, err)
+	}
+	lastmod := stat.ModTime()
+	go func() {
+		fmt.Println("Watching for changes in the binary")
+		ticker := time.NewTicker(5 * time.Second)
```

Comment:
why not https://pkg.go.dev/github.com/fsnotify/fsnotify#section-readme

Reply:
Over complicated for the task at hand IMO.